src/kakarot/gas.cairo

```diff
@@ -101,4 +101,57 @@ namespace Gas {
 
         return memory_expansion_cost(words_len, offset.low + size.low);
     }
+
+    // @notice given two memory chunks, compute the maximum expansion cost
+    //  based on the maximum offset reached by each chunks.
+    // @param words_len The current length of the memory as felt252.
+    // @param offset_1 The offset of the first memory chunk as Uint256.
+    // @param size_1 The size of the first memory chunk as Uint256.
+    // @param offset_2 The offset of the second memory chunk as Uint256.
+    // @param size_2 The size of the second memory chunk as Uint256.
+    // @return cost The expansion gas cost for chunk who's ending offset is the largest.
+    func max_memory_expansion_cost{range_check_ptr}(
+        words_len: felt, offset_1: Uint256, size_1: Uint256, offset_2: Uint256, size_2: Uint256
```

Comment:
you'll be able to check with a unit test and `--profile-cairo` but using pointers in the signature should reduce the steps (one less per arg, so 4 less in total). More out of curiosity than a real game changer though